### PR TITLE
Add CI migration blog post

### PR DESCRIPTION
**Pull request by snadukudy**
Add blog post on CircleCI migration to GitHub Actions

### CODE REVIEW

File: website/blog/2024-07-25-ci-migration.mdx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+## Velox’s Continuous Integration Workload
+
+Continuous Integration (CI) is crucial for Velox’s success as an open source project as it helps protect from bugs and errors, reduces likelihood of conflicts and leads to increased community trust in the project. This is to ensure the Velox builds works well on a myriad of system architectures, operating systems and compilers - along with from the ones used internally at Meta. The OSS build version of Velox also supports additional features that aren't used internally in Meta (for example, support for Cloud blob stores, etc.).
+
+When a pull request is submitted to Velox, the following jobs are executed:
+
+1. Linting and Formatting workflows:
+    a. Header checks
+    b. License checks
+    c. Basic Linters
```
Comment (assignUser) [inline website/blog/2024-07-25-ci-migration.mdx:25]: Ordered lists can only be numeric. I think an unordered list (-/*) is sufficient from a contents perspective as well.
image

File: website/blog/2024-07-25-ci-migration.mdx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    a. Header checks
+    b. License checks
+    c. Basic Linters 
+2. Ensure Velox builds on various platforms
+    a. MacOS (13/14, M1)
+    b. Linux (Ubuntu/Centos)
+3. Ensure Velox builds under its various configurations
+    a. Debug / Release builds
+    b. Build default Velox build
+    c.Build Velox with support for Parquet/Arrow/ External Adapters (S3/HDFS/GCS etc.)
```
Comment (assignUser) [inline website/blog/2024-07-25-ci-migration.mdx:32]:
```suggestion
    c. Build Velox with support for Parquet, Arrow and External Adapters (S3/HDFS/GCS etc.)
```

File: website/blog/2024-07-25-ci-migration.mdx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+Continuous Integration (CI) is crucial for Velox’s success as an open source project as it helps protect from bugs and errors, reduces likelihood of conflicts and leads to increased community trust in the project. This is to ensure the Velox builds works well on a myriad of system architectures, operating systems and compilers - along with from the ones used internally at Meta. The OSS build version of Velox also supports additional features that aren't used internally in Meta (for example, support for Cloud blob stores, etc.).
+
+When a pull request is submitted to Velox, the following jobs are executed:
+
+1. Linting and Formatting workflows:
+    a. Header checks
+    b. License checks
+    c. Basic Linters 
+2. Ensure Velox builds on various platforms
+    a. MacOS (13/14, M1)
```
Comment (assignUser) [inline website/blog/2024-07-25-ci-migration.mdx:27]:
```suggestion
    a. MacOS (Intel, M1)
```

File: website/blog/2024-07-25-ci-migration.mdx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ <img src="/img/abstract-white-background.jpg" height= "100%" width="100%"/>
+</figure>
+
+## TL;DR
+
+In late 2023, the Meta OSS (Open Source Software) Team requested all Meta teams to move the CI deployments from CircleCI to Github Actions. [Voltron Data](http://voltrondata.com) and Meta in collaboration migrated all the deployed Velox CI jobs. For the year 2024, Velox CI spend was on track to overshoot the allocated resources by a considerable amount of money. As part of this migration effort, the CI workloads were consolidated and optimized by Q2 2024, bringing down the projected 2024 CI spend by 51%.
+
+## Velox’s Continuous Integration Workload
+
+Continuous Integration (CI) is crucial for Velox’s success as an open source project as it helps protect from bugs and errors, reduces likelihood of conflicts and leads to increased community trust in the project. This is to ensure the Velox builds works well on a myriad of system architectures, operating systems and compilers - along with from the ones used internally at Meta. The OSS build version of Velox also supports additional features that aren't used internally in Meta (for example, support for Cloud blob stores, etc.).
```
Comment (assignUser) [inline website/blog/2024-07-25-ci-migration.mdx:18]:
```suggestion
Continuous Integration (CI) is crucial for Velox’s success as an open source project as it helps protect from bugs and errors, reduces likelihood of conflicts and leads to increased community trust in the project. This is to ensure the Velox builds works well on a myriad of system architectures, operating systems and compilers - along with the ones used internally at Meta. The OSS build version of Velox also supports additional features that aren't used internally in Meta (for example, support for Cloud blob stores, etc.).
```

File: website/blog/2024-07-25-ci-migration.mdx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        i. Ubuntu Linux
+        ii. Centos
+        iii. Presto Linux image
+6. Documentation build and publish Job
+    a. If underlying documentation is changed, Velox documentation pages are rebuilt and published
+    b. Netlify is used for publishing Velox web pages
+
+## Velox CI Optimization
+
+Previous implementation of CI in CircleCI grew organically and was unoptimized, resulting in long build times and also, significantly costlier. This opportunity to migrate to Github Actions helped to take a holistic view of CI deployments and actively optimized to reduce build times and CI spend. Note that however there has been continued investment in increasing test times to further increase Velox reliability and stability. Some of the optimizations completed are below:-
```
Comment (assignUser) [inline website/blog/2024-07-25-ci-migration.mdx:52]:
```suggestion
Previous implementation of CI in CircleCI grew organically and was unoptimized, resulting in long build times, and also significantly costlier. This opportunity to migrate to Github Actions helped to take a holistic view of CI deployments and actively optimized to reduce build times and CI spend. Note however, that there has been continued investment in reducing test times to further improve Velox reliability, stability and developer experience. Some of the optimizations completed are:  
```

File: website/blog/2024-07-25-ci-migration.mdx [hunk 6]
```diff
@@ -0,0 +1,78 @@
+---
+slug: ci-migration
+title: "Optimizing and Migrating Velox CI Workloads to Github Actions"
+authors: [jwujciak, kgpai]
+tags: [tech-blog, packaging]
+---
+
+<figure>
+ <img src="/img/abstract-white-background.jpg" height= "100%" width="100%"/>
+</figure>
```
Comment (assignUser) [inline website/blog/2024-07-25-ci-migration.mdx:10]: While a nice image, I am not sure it really adds anything?

Comment (snadukudy) [reply website/blog/2024-07-25-ci-migration.mdx:10]: yep, that's correct. We can remove it as well. Please let me know.

File: website/blog/2024-07-25-ci-migration.mdx (hunk 4 above)
Comment (snadukudy) [reply website/blog/2024-07-25-ci-migration.mdx:18]: thanks, updated now

File: website/blog/2024-07-25-ci-migration.mdx (hunk 2 above)
Comment (snadukudy) [reply website/blog/2024-07-25-ci-migration.mdx:32]: thanks, updated

File: website/blog/2024-07-25-ci-migration.mdx (hunk 3 above)
Comment (snadukudy) [reply website/blog/2024-07-25-ci-migration.mdx:27]: thanks, updated

File: website/blog/2024-07-25-ci-migration.mdx (hunk 5 above)
Comment (snadukudy) [reply website/blog/2024-07-25-ci-migration.mdx:52]: thanks, updated

File: website/blog/2024-07-25-ci-migration.mdx (hunk 1 above)
Comment (snadukudy) [reply website/blog/2024-07-25-ci-migration.mdx]: thanks, updated

**Review by kgpai — APPROVED** [review]
LGTM.

**Comment by kgpai** [comment]
@snadukudy This doesnt have to be a draft anymore.

File: website/blog/2024-07-25-ci-migration.mdx (hunk 6 above)
Comment (snadukudy) [reply website/blog/2024-07-25-ci-migration.mdx:10]: cc @kgpai for thoughts

**Comment by kgpai** [comment]
Got verbal approval from Sridhar.

**Pull request closed without merging**